dot-base/fhir-server
# feat: observation scores auto-generation

# Ticket 🎫

This closes #259.

# Description 📖
This adds an interceptor to the FHIR server that (re)creates Observations for each expression item annotated with a terminology code on a newly created or updated QuestionnaireResponse.

# How to Test? 🧪

Run the server and post a patient and a QuestionnaireResponse to the server via Postman. Get the Observation endpoint and check that the correct Observations were created.

# Is there something controversial ? 🚨

- Deletion is not supported by the interceptor.
- Observations always have the status `final`. Perhaps we should use a different status after an update to the response, e.g. `amended`.
- Should the helper functions of the interceptor be members, static or moved to a util?

# Checklist ✅
